### ISSUE TITLE
Decouple requests from OAuth store

### ISSUE THREAD

**Issue by nearengine** [synapsestudios/lively#69]
Currently, requests are handled through the OAuth store. This causes two issues: for projects that don't utilize OAuth, there is no way to use another form of authentication, since all requests depend on it. 

Second, even if the OAuth object is set to "false" in the configuration, we're still passing an OAuth object around so there's no clean way to detect on the frontend if OAuth is enabled for a given API or not (ie, to hide the lock button when OAuth is not being used). 

Basically we just need to de-couple requests from OAuth. 

Relevant files include oauth2.js (where the store is generated from config), api.jsx (where the oauth store gets passed to the components), and config.github.js (where the config details for the oauth and requests are stored). 

Another thing to look into is the oauth2.type configuration for APIs, which appears to never be used in the app.

**Comment by zpchavez** [comment]
Is this still a problem, post flux?

**Comment by nearengine** [comment]
Nope, I think the refactor cleaned this up a lot

**Comment by zpchavez** [comment]
:boom:
